pkg/gameservers/portallocator.go

```diff
@@ -83,31 +91,27 @@ func NewPortAllocator(minPort, maxPort int32,
 		nodeSynced:         nodes.Informer().HasSynced,
 	}
 	pa.logger = runtime.NewLoggerWithType(pa)
+	pa.workerqueue = workerqueue.NewWorkerQueue(pa.syncPorts, pa.logger, stable.GroupName+".PortAllocator")
 
 	pa.gameServerInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		DeleteFunc: pa.syncDeleteGameServer,
 	})
 
 	// Experimental support for node adding/removal
 	pa.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: pa.syncAddNode,
+		AddFunc: func(obj interface{}) {
+			node := obj.(*corev1.Node)
```

Comment:
I don't know the semantics of the event handler, but is there a chance that `obj` isn't of `*corev1.Node` type? In other words, is this a safe type assertion?

Reply:
A good question - basically, there isn't really. But a good question!

If something other than a Node comes through a Node event handler, then something _very_ weird has happened -- at which point, it really should panic.

Here another example in the Kubernetes sample-controller doing similar things:
https://github.com/kubernetes/sample-controller/blob/master/controller.go#L129

Reply:
Then panicking is exactly the right thing to do.